misc/dbt-materialize/dbt/include/materialize/macros/deploy/deploy_init.sql

```diff
@@ -177,13 +179,141 @@
             {{ run_query(create_cluster) }}
             {{ set_cluster_ci_tag() }}
+            {{ internal_copy_cluster_grants(schema, deploy_schema) }}
         {% endif %}
     {% endif %}
 {% endfor %}
 {% endmacro %}
+
+{% macro internal_copy_schema_default_privs(from, to) %}
+
+{% set find_default_privs %}
+SELECT
+  'ALTER DEFAULT PRIVILEGES ' ||
+  CASE
+    WHEN object_owner = 'PUBLIC' THEN 'FOR ALL ROLES '
+    ELSE 'FOR ROLE ' || object_owner
+  END ||
+  'IN SCHEMA {{ to }} '         ||
+  'GRANT '  || privilege_type   || ' ' ||
+  'ON '     || object_type      || 's ' ||
+  'TO '     || grantee
+FROM mz_internal.mz_show_default_privileges
+WHERE database = current_database() AND schema = {{ dbt.string_literal(from) }}
+    AND object_owner <> 'none' AND grantee <> 'none'
+{% endset %}
+
+{% set alter_defaults = run_query(find_default_privs) %}
+{% if execute %}
+    {% for alter in alter_defaults.rows %}
+        {{ run_query(alter) }}
+    {% endfor %}
+{% endif %}
+{% endmacro %}
+
+{% macro internal_copy_schema_grants(from, to) %}
+{% set find_revokes %}
+WITH schema_privledge AS (
+    SELECT mz_internal.mz_aclexplode(s.privileges).*
+    FROM mz_schemas s
+    JOIN mz_databases d ON s.database_id = d.id
+    WHERE d.name = current_database()
+        AND s.name = {{ dbt.string_literal(to) }}
+)
+
+SELECT 'REVOKE '    || s.privilege_type || ' ' ||
+       'ON SCHEMA ' || quote_ident({{ dbt.string_literal(to) }}) || ' ' ||
+       'FROM '      || quote_ident(grantee.name)
+FROM schema_privledge AS s
+JOIN mz_roles AS grantee ON s.grantee = grantee.id
+WHERE grantee.name NOT IN ('none', 'mz_system', 'mz_support', current_role)
+{% endset %}
+
+{% set revokes = run_query(find_revokes) %}
+{% if execute %}
+    {% for revoke in revokes.rows %}
+        {{ run_query(revoke[0]) }}
+    {% endfor %}
+{% endif %}
+
+{% set find_grants %}
+WITH schema_privledge AS (
+    SELECT mz_internal.mz_aclexplode(s.privileges).*
+    FROM mz_schemas s
+    JOIN mz_databases d ON s.database_id = d.id
+    WHERE d.name = current_database()
+        AND s.name = {{ dbt.string_literal(from) }}
+)
+
+SELECT 'GRANT '     || s.privilege_type || ' ' ||
+       'ON SCHEMA ' || quote_ident({{ dbt.string_literal(to) }}) || ' ' ||
+       'TO '        || quote_ident(grantee.name)
+FROM schema_privledge AS s
+JOIN mz_roles AS grantee ON s.grantee = grantee.id
+WHERE grantee.name NOT IN ('none', 'mz_system', 'mz_support', current_role)
+{% endset %}
+
+{% set grants = run_query(find_grants) %}
+{% if execute %}
+    {% for grant in grants.rows %}
+        {{ run_query(grant[0]) }}
+    {% endfor %}
+{% endif %}
+
+{% endmacro %}
+
+{% macro internal_copy_cluster_grants(from, to) %}
+{% set find_revokes %}
+WITH schema_privledge AS (
+    SELECT mz_internal.mz_aclexplode(s.privileges).*
+    FROM mz_schemas s
+    JOIN mz_databases d ON s.database_id = d.id
+    WHERE d.name = current_database()
+        AND s.name = {{ dbt.string_literal(to) }}
+)
```

Comment:
This doesn't look right! Don't you want to be listing cluster privileges here?

Reply:
You are right, and I'm concerned my test passed. I'll figure that out.